### PR TITLE
Aborting of all discovery immediately

### PR DESCRIPTION
**Pull request by Sanan07**
## Description
Abort all ongoing discovery if aborting was requested. Fix aborting flow.

### CODE REVIEW

**Comment by nohwnd** [comment]
I saw that there are changes to public api. Want to revisit this before merge.

**Review by Haplois — CHANGES_REQUESTED** [review]
Some comments on modified public APIs.

File: src/Microsoft.TestPlatform.Common/Interfaces/Engine/ClientProtocol/IProxyDiscoveryManager.cs [hunk 1]
```diff
@@ -22,11 +22,12 @@ public interface IProxyDiscoveryManager
...
         void DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEventsHandler2 eventHandler);
-
+     
         /// <summary>
-        /// Aborts the test operation.
+        /// Aborts the discovery operation
         /// </summary>
-        void Abort();
+        /// <param name="eventHandler"> EventHandler for handling discovey opearation from Engine</param>
+        void Abort(ITestDiscoveryEventsHandler2 eventHandler);
```
[the remark this answers is not in the document]
Comment (Sanan07) [reply src/Microsoft.TestPlatform.Common/Interfaces/Engine/ClientProtocol/IProxyDiscoveryManager.cs:30]: Basically, we introduced new aborting message, so in old VS there is no that message, so testhost will not handle it. In new VS we will send that message, but testhost will not handle it. When I tested I used 16.5 Test.Sdk and it worked fine, but will test in different scenarious as well.

**Comment by Sanan07** [comment]
Closing this as will create new PR with small change.

**Comment by nohwnd** [comment]
Why though? PR is a pointer to a branch, you could have just pushed the change to this branch.

**Pull request closed without merging**